controllers/constraintstatus_controller.go

```diff
@@ -0,0 +1,349 @@
+	log.Info("Reconciling ConstraintPodStatus and Config")
+	// This is used for RequeueAfter
+	requeueTime := time.Second * 0
+
+	// Get config or create if not exist
+	config := &v1alpha1.Config{}
+	err := r.Get(ctx, types.NamespacedName{
+		Namespace: r.Namespace,
+		Name:      "config",
+	}, config)
+
+	if apierrors.IsNotFound(err) {
+		config = &v1alpha1.Config{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "config",
+				Namespace: r.Namespace,
+			},
+		}
+
+		err = r.Create(ctx, config)
+		if err != nil {
+			log.Error(err, "Fail to Create Config resource, requeue")
+
+			return reconcile.Result{}, err
+		}
+
+		log.Info("The Gatekeeper's Config resource is just created")
+	}
+
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	constraintPodStatus := &v1beta1.ConstraintPodStatus{}
+
+	err = r.Get(ctx, request.NamespacedName, constraintPodStatus)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.V(1).Info("Cannot find any constraintPodStatus")
+
+			err = r.handleDeleteEvent(ctx, request.Name, config)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			return reconcile.Result{}, nil
+		}
+		// Requeue
+		return reconcile.Result{}, err
+	}
+
+	labels := constraintPodStatus.GetLabels()
+
+	constraintKind := labels["internal.gatekeeper.sh/constraint-kind"]
+	constraintName := labels["internal.gatekeeper.sh/constraint-name"]
+
+	constraintGVR := schema.GroupVersionResource{
+		Group:    "constraints.gatekeeper.sh",
+		Version:  "v1beta1",
+		Resource: strings.ToLower(constraintKind),
+	}
+
+	constraint, err := r.DynamicClient.Resource(constraintGVR).Get(ctx, constraintName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			r.Log.Info("The Constraint is founded in the config resource", "constraintName:", constraint.GetName())
+
+			return reconcile.Result{}, nil
+		}
+
+		return reconcile.Result{}, err
+	}
+
+	constraintMatchKinds, _, err := unstructured.NestedSlice(constraint.Object, "spec", "match", "kinds")
+	if err != nil {
+		r.Log.V(1).Info("There is no provided kinds in contsraint", "constraintName:", constraint.GetName())
+
+		return reconcile.Result{}, nil //nolint:nilerr
+	}
+
+	constraintSyncOnlyEntries, err := r.getSyncOnlys(constraintMatchKinds)
+	if err != nil {
+		if errors.Is(err, ErrNotFoundDiscovery) {
+			r.Log.V(1).Info("Cannot find matched discovery. Requeue after 3 mins")
+
+			requeueTime = time.Minute * 3
+		} else {
+			log.Error(err, "Error to get matching kind and apigroup")
+
+			return reconcile.Result{}, err
+		}
+	}
+
+	r.addConstraintPodStatusToMap(request.Name, *constraintSyncOnlyEntries)
+	updatedSyncOnly := r.getUniqSyncOnly()
+
+	err = r.updateConfigSyncOnlyEntry(ctx, config, updatedSyncOnly)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if requeueTime != time.Second*0 {
+		return reconcile.Result{RequeueAfter: requeueTime}, nil
+	}
+
+	return reconcile.Result{}, nil
```

Comment:
`requeueTime` will either be set or nil, so no need to guard against using it in the return.
```suggestion
	return reconcile.Result{RequeueAfter: requeueTime}, nil
```